src/algorithm/nest_cfgs.rs

```diff
@@ -43,21 +43,23 @@ use std::hash::Hash;
 
 use itertools::Itertools;
 
-use crate::hugr::views::HugrView;
+use crate::hugr::rewrite::outline_cfg::OutlineCfg;
+use crate::hugr::views::sibling::SiblingMut;
+use crate::hugr::views::{HierarchyView, HugrView, SiblingGraph};
+use crate::hugr::{HugrMut, Rewrite};
+use crate::ops::handle::{BasicBlockID, CfgID};
 use crate::ops::OpTag;
 use crate::ops::OpTrait;
-use crate::{Direction, Node};
-
-// TODO: transform the CFG: each SESE region can be turned into its own Kappa-node
-// (in a BB with one predecessor and one successor, which may then be merged
-//     and contents parallelized with predecessor or successor).
-
-/// A view of a CFG. `T` is the type of basic block; one interpretation of `T` would be a BasicBlock
-/// (e.g. `Node`) in the Hugr, but this extra level of indirection allows "splitting" one HUGR BB
-/// into many (or vice versa). Since SESE regions are bounded by edges between pairs of such `T`, such
-/// splitting may allow the algorithm to identify more regions than existed in the underlying CFG
+use crate::{Direction, Hugr, Node};
+
+/// A "view" of a CFG in a Hugr which allows basic blocks in the underlying CFG to be split into
+/// multiple blocks in the view (or merged together).
+/// `T` is the type of basic block; this can just be a BasicBlock (e.g. [`Node`]) in the Hugr,
+/// for an [IdentityCfgMap] if the extra level of indirection is not required. However, since
+/// SESE regions are bounded by edges between pairs of such `T`, such splitting may allow the
+/// algorithm to identify more regions than existed in the underlying CFG
 /// (without mutating the underlying CFG perhaps in vain).
-pub trait CfgView<T> {
+pub trait CfgNodeMap<T> {
```

Comment:
I'm renaming this as a could-be-separate part of the PR ;). While I'm about it, I could make the `T` a `type` member of the trait (`NodeType`, say)...